src/index.js

```diff
@@ -15,6 +15,17 @@ const isLocalhost = () => Boolean(
     )
 )
 
+let waitWindowLoad
+// Typically, a browser that supports `serviceWorker` should also have supported
+// `Promise`. But as this package can be used in environments without service
+// worker support (in that case it would do nothing), there's a chance that
+// `Promise` does not exist. So we must check for its existence first.
+if (typeof Promise !== 'undefined') {
+  waitWindowLoad = new Promise(resolve => window.addEventListener('load', resolve))
+} else {
+  waitWindowLoad = { then: (cb) => window.addEventListener('load', cb) }
+}
```

Comment:
These code reference `window` in the outter scope, which cause potential breaking changes.

For example, it breaks the SSR process in VuePress https://github.com/vuejs/vuepress/issues/2222

@sodatea

Reply:
Thanks. I'll fix it soon.